framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/jobs/JobManager.java

```diff
@@ -26,34 +38,62 @@ public void removeJobListener(IJobListener l) {
 	}
 
 	@Override
-	public void addJob(IJob job) {
+	public IJob scheduleJob(String jobName, IRunnableWithProgress runnable) {
+		SimpleJob job = new SimpleJob(jobName, generateJobId(), runnable);
 		currentJobs.put(job.getId(), job);
-		ArrayList<IJobListener> tmp = new ArrayList<>(listeners);
-		for( IJobListener l : tmp ) {
-			l.jobAdded(job);
-		}
+		fireJobAdded(job);
+		schedule(job);
+		return job;
 	}
 
 	@Override
-	public void removeJob(IJob job, IStatus status) {
-		currentJobs.remove(job.getId());
+	public IJob scheduleJob(String jobName, IStatusRunnableWithProgress runnable) {
+		SimpleJob job = new SimpleJob(jobName, generateJobId(), runnable);
+		currentJobs.put(job.getId(), job);
```

Comment:
what if a job with the same id already exists? It would get evicted off the map and there would be no way to access that instance any more. I'm aware that we're making sure that the id is unique but I still believe that we should at least throw here if the job already exists, just in case a latter change would be flawed and have non-unique ids.